### DISCUSSION TITLE
Alias rules?

### DISCUSSION THREAD

**Discussion by hillin** [pest-parser/pest#677] [Q&A] [answered]
e.g.
```
foo = { "foo" }
bar = { "bar" }
foo_or_bar = { foo | bar }
parser = { "abc" ~ foo_or_bar ~ "def" }
```

I'd like to have the `foo` and `bar` rules in the parse result of `parser`, but not `foo_or_bar`, i.e. `foo_or_bar` is just an alias of `foo | bar`.

**Comment by glyn** [comment, marked as answer]
How about using a [silent rule](https://pest.rs/book/grammars/syntax.html#silent-and-atomic-rules)?

**Reply by hillin** [reply]
Oh, I thought silent rule omits its inner tokens too. Thanks!

**Reply by glyn** [reply]
I can see why you thought that based on the documentation. Any inner rules of a silent rule are not themselves silent (unless you declare them to be silent). Perhaps it's worth raising a doc issue?

**Reply by hillin** [reply]
You bet. I've opened an issue here: https://github.com/pest-parser/pest/issues/678
